File: PennyPincher.API/PennyPincher/Models/CashFlowDto.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace PennyPincher.Models
{

    public enum FlowTypes
    {
        income = 0,
        expense= 1
    }
    public class CashFlowDto : CashFlowUpdateDto
    {
        [Required(ErrorMessage ="An integer ID is required for this CashFlow")]
        [Range(0,int.MaxValue)]
        public int Id { get; set; }

    }
}

File: PennyPincherUI/src/app/models/cashflowdto.ts
import type { CATEGORY_TYPES } from "./expense";

export class CashFlowDto {
	constructor(
		public id: number,
		public name: string,
		public description: string,
		public amount: number,
		public flow: FLOW_TYPES,
		public category: CATEGORY_TYPES,
	) {}
}

export class CashFlowUpdateDto {
	constructor(
		public name: string,
		public description: string,
		public amount: number,
		public flow: FLOW_TYPES,
		public category: CATEGORY_TYPES,
	) {}
}

export enum FLOW_TYPES {
	income = 0,
	expense = 1,
}
